backend/substrapp/ledger/connection.py

```diff
@@ -119,11 +119,25 @@ def _get_hfc(channel_name):
 
     results = _deserialize_discovery(results)
 
+    _validate_channels(channel_name, results)
     _update_client_with_discovery(client, results)
 
     return loop, client, user
 
 
+def _validate_channels(channel_name, discovery_results):
+
+    channel = [c for c in settings.LEDGER_CHANNELS if c['name'] == channel_name][0]
+
+    # Esnure solo channels have 1 member at most
+    if channel_name.startswith('solo-') or channel['restricted']:
+        channel_members = list(discovery_results['config']['msps'].keys())
```

Comment:
You can use the `results` from `_deserialize_discovery(results)` which already deserialize peer members  in `results['members']` :)

Reply:
I can't use that method because the `members` value contains the peers that have **joined** the channel (in this case only peer-1)

```
[[{'mspid': 'MyOrg1MSP', 'id_bytes': '-----BEGIN CERTIFICATE-----\nMIIC0DCCAnegAwIBAgIUA0DXmM3Y0oH+ZgQuQiXrIZlcs04wCgYIKoZIzj0EAwIw\nbjELMAkGA1UEBhMCRlIxGTAXBgNVBAgTEExvaXJlLUF0bGFudGlxdWUxDzANBgNV\nBAcTBk5hb │nRlczEQMA4GA1UEChMHU3Vic3RyYTEPMA0GA1UECxMGRmFicmljMRAw\nDgYDVQQDEwdyY2FPcmcxMB4XDTIwMTAxNDE2MDQwMFoXDTIxMTAxNDE2MDkwMFow\nWjELMAkGA1UEBhMCVVMxFzAVBgNVBAgTDk5vcnRoIENhcm9saW5hMRQwEgYDVQQK\nEwtIeXBlcmxlZGdlcjENMAsGA1UECxMEcGV │lcjENMAsGA1UEAxMEdXNlcjBZMBMG\nByqGSM49AgEGCCqGSM49AwEHA0IABKdE5vumVaXCRq6gTDcBMXnkwpz3cCAhc7fo\nH4fKdk8lUdK/z6Q+Bn9p3f5i8Nzg36IzdMBOfMygmAsiowkjKFqjggEFMIIBATAO\nBgNVHQ8BAf8EBAMCB4AwDAYDVR0TAQH/BAIwADAdBgNVHQ4EFgQUz8uyUuFUi │Pxn\nVXw0hlTCmhRQnvUwHwYDVR0jBBgwFoAUZqHx0bgfq6o42oD4tGd9XOnP71cwSgYD\nVR0RBEMwQYI/bmV0d29yay1vcmctMS1wZWVyLTEtaGxmLWs4cy1lbnJvbGxtZW50\nLW9wZXJhdG9yLTY0ZmY3OTZiN2Z3NnM5MFUGCCoDBAUGBwgBBEl7ImF0dHJzIjp7\nImhmLkFmZmlsaWF0aW9uI │joiIiwiaGYuRW5yb2xsbWVudElEIjoidXNlciIsImhm\nLlR5cGUiOiJwZWVyIn19MAoGCCqGSM49BAMCA0cAMEQCIFxvWtIazap7Cl74yVi2\nPhWju2WqpQ5c20duPC0f+R8gAiAcd1OaUBRNWq2nBbMPQsfIwbCRZzvjm+OvPzBh\nnWyopg==\n-----END CERTIFICATE-----\n', 'ledger │_height': 3, 'chaincodes': [{'name': '_lifecycle', 'version': '1'}], 'endpoint': 'network-org-1-peer-1-hlf-peer.org-1:7051'}], []]
```

While `discovery_results['config']['msps'].keys()` contains all the entities allowed on the channel:

```
['MyOrdererMSP', 'MyOrg2MSP', 'MyOrg1MSP']    
```

We want to raise an error when the channel is configured to allow multiple users (even if only one user has joined the channel)

Reply:
Okay, thanks 👍